src/esp32/webhook.rs

```diff
@@ -0,0 +1,94 @@
+#[allow(unused_imports)] // need this trait for `get_attribute`
+use crate::common::config::Component;
+use crate::common::config::DynamicComponentConfig;
+use crate::proto::app::v1::ConfigResponse;
+use embedded_svc::{
+    http::{client::Client as HttpClient, Method},
+    io::Write,
+};
+use esp_idf_svc::http::client::{Configuration as HttpConfiguration, EspHttpConnection};
+use log::*;
+use serde_json::json;
+use thiserror::Error;
+
+#[derive(Debug, Error, PartialEq)]
+pub enum WebhookError {
+    #[error("`webhook` not present in board attributes")]
+    NoWebhook,
+    #[error("`webhook-secret` not present in board attributes")]
+    NoWebhookSecret,
+    #[error("Error with webhook request: {0}")]
+    RequestError(String),
+    #[error("Error in config: {0}")]
+    ConfigError(String),
+}
+
+/// Checks the board config for a `webhook` and `webhook-secret` attribute, sending a GET
+/// request with credentials for an SDK to make a connection.
+pub fn handle_webhook(config: ConfigResponse) -> Result<(), WebhookError> {
+    let config = config
+        .config
+        .as_ref()
+        .ok_or_else(|| WebhookError::ConfigError("board does not have config".to_string()))?;
+    let components = &config.components; // component config
+    let cloud = config.cloud.as_ref().ok_or_else(|| {
+        WebhookError::ConfigError("board config does not have cloud config".to_string())
+    })?;
+    let fqdn = &cloud.fqdn; // robot's url
+    let board_cfg: DynamicComponentConfig = components
+        .iter()
+        .find(|x| x.r#type == "board")
+        .ok_or_else(|| {
+            WebhookError::ConfigError("board component not found in config".to_string())
+        })?
+        .try_into()
+        .map_err(|_| {
+            WebhookError::ConfigError(
+                "could not convert board to DynamicComponentConfig".to_string(),
```

Comment:
this should really say "could not convert board config to DynamicComponentConfig". Since we're not dealing with an actual board at this stage we want to avoid confusion in the wording